### PR TITLE
Initial unix domain socket support

### PR DESCRIPTION
**Pull request by paul-guo-**
I will add test cases if needed.

### CODE REVIEW

File: src/common/comm_server.c [hunk 1]
```diff
@@ -51,6 +54,62 @@ int start_listener() {
 }
 
 /*
+ * Functoin binds the socket and starts listening on it: unix domain socket.
```
Comment (davecramer) [inline src/common/comm_server.c]: s/b Function

File: src/common/comm_server.c [hunk 2]
```diff
@@ -51,6 +54,62 @@ int start_listener() {
...
+#endif
+    if (listen(sock, 10) == -1) {
+        lprintf(ERROR, "Cannot listen the socket: %s", strerror(errno));
+    }
+
+    return sock;
+}
+
+/*
  * Fuction waits for the socket to accept connection for finite amount of time
```
Comment (davecramer) [inline src/common/comm_server.c]: Different spelling error for Fuction

File: src/plc_configuration.c [hunk 3]
```diff
@@ -378,20 +392,39 @@ char *get_sharing_options(plcContainerConf *conf) {
             totallen += strlen(volumes[i]);
         }
 
-        res = palloc(totallen + 2 * conf->nSharedDirs);
+		if (!conf->isNetworkConnection) {
```
Comment (davecramer) [inline src/plc_configuration.c]: indention ??

**Comment by paul-guo-** [comment]
@davecramer typo were fixed. Thanks.

For indentation, I think we should follow the pg convention which seems to be using tab (with 4 space wideth) for indentation. Later we might replace all other space indentation so during development we will use tab for new code change.

File: src/common/comm_connectivity.c [hunk 4]
```diff
@@ -370,14 +376,67 @@ plcConn *plcConnect(int port) {
...
+    tv.tv_sec  = 0;
+    tv.tv_usec = 500000;
+    setsockopt(sock, SOL_SOCKET, SO_RCVTIMEO, (char *)&tv, sizeof(struct timeval));
+
+    result = plcConnInit(sock);
+
+    return result;
+}
+
+
```
[the remark this answers is not in the document]
Comment (paul-guo-) [reply src/common/comm_connectivity.c]: Done. Thanks.

**Comment by paul-guo-** [comment]
@stanlyxiang 

I changed the code so there is no need to fix R code. Also memset the conf struct so use_network has an explicit value. Fixed another small bug additionally.

Thanks for reviewing.

**Comment by paul-guo-** [comment]
I'll merge them at first. Currently pg_regress runs with this setting by default. There will have some extra security related things need to do next.

**Pull request merged**